### PR TITLE
implemented basic DSL, `hooks` and `validate`

### PR DESCRIPTION
**Pull request by paa001**
that commit contains #168 and #169 issues

### CODE REVIEW

File: lib/generamba/dsl/methods.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+module Generamba
+  module DSL
+    module Methods
```
Comment (etolstoy) [inline lib/generamba/dsl/methods.rb:3]: What do you think of adding logging capability? Something like `log(message, log_level)`, where each `log_level` has a corresponding text color.

File: lib/generamba/rake/application.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    @error_hooks  = {}
+  end
+
+  def standard_rake_options
+    ramba_version = [
+      '--version', '-V',
+      'Display the program version.',
+      lambda do |_|
+        puts "ramba, version v.#{Generamba::VERSION}"
+        puts "rake, version #{RAKEVERSION}"
```
Comment (etolstoy) [inline lib/generamba/rake/application.rb]: Why do you want to print rake version?

File: lib/generamba/version.rb [hunk 3]
```diff
@@ -1,5 +1,6 @@
 module Generamba
-  VERSION = '1.3.0'
-  RELEASE_DATE = '05.10.2016'
-  RELEASE_LINK = "https://github.com/rambler-digital-solutions/Generamba/releases/tag/#{VERSION}"
+  VERSION = '2.0.0'.freeze
```
Comment (etolstoy) [inline lib/generamba/version.rb:2]: So, we really decided to deprecate Ruby 2.2?

File: spec/dsl/validators_spec.rb [hunk 4]
```diff
@@ -0,0 +1,25 @@
+describe Generamba::DSL::Validators do
+  describe '#validate' do
+    context 'when has block' do
+      it 'should raise error if it retutn false' do
```
Comment (etolstoy) [inline spec/dsl/validators_spec.rb]: Typo :)

File: spec/dsl/validators_spec.rb [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+        expect {
+          validate :test_validator do
+            false
+          end
+        }.to raise_error(RuntimeError)
+      end
+
+      it 'should remain silent if it retutn true' do
```
Comment (etolstoy) [inline spec/dsl/validators_spec.rb]: Another one typo in word `retutn`

**Review by etolstoy — APPROVED** [review]
Looks great, thanks for your work! :)

File: lib/generamba/dsl/methods.rb (hunk 1 above)
Comment (paa001) [reply lib/generamba/dsl/methods.rb:3]: I think it's good idea. We can do it in another issue.

File: lib/generamba/rake/application.rb (hunk 2 above)
Comment (paa001) [reply lib/generamba/rake/application.rb]: Because it is only a new DSL for rake :)

File: lib/generamba/version.rb (hunk 3 above)
Comment (paa001) [reply lib/generamba/version.rb:2]: Why? Specs were successful for 2.2

File: spec/dsl/validators_spec.rb (hunk 4 above)
Comment (paa001) [reply spec/dsl/validators_spec.rb]: Sorry :)

File: spec/dsl/validators_spec.rb (hunk 5 above)
Comment (paa001) [reply spec/dsl/validators_spec.rb]: Oh)) It is friday :)

Comment (paa001) [reply spec/dsl/validators_spec.rb]: fixed

File: lib/generamba/rake/application.rb (hunk 2 above)
Comment (etolstoy) [reply lib/generamba/rake/application.rb]: But does it matter for the end user? :)

File: lib/generamba/version.rb (hunk 3 above)
Comment (etolstoy) [reply lib/generamba/version.rb:2]: I thought that `freeze` first appeared in 2.3

Comment (paa001) [reply lib/generamba/version.rb:2]: No, method `freeze` was addded a long time ago.

File: lib/generamba/rake/application.rb (hunk 2 above)
Comment (paa001) [reply lib/generamba/rake/application.rb]: I think, it is not very important :) Ok, i remove it)

Comment (paa001) [reply lib/generamba/rake/application.rb]: done

**Comment by etolstoy** [comment]
I think it's ready for merging :)

**Comment by paa001** [comment]
Who will do merging? ))

**Comment by etolstoy** [comment]
@paa001 I think you can do it :)

**Pull request merged**